src/main/views/includes/macros/partyDetails.njk

```diff
@@ -1,61 +1,81 @@
-{% from "includes/macros/form.njk" import csrfProtection, textInput, radioGroup, saveAndContinueButton %}
+{% from "includes/macros/form.njk" import readOnlyField, csrfProtection, textInput, radioGroup, saveAndContinueButton %}
 {% from "includes/macros/address.njk" import addressFragment, correspondenceAddressFragment %}
+{% from "includes/macros/table.njk" import tableStart,tableEnd, row, rowWithMultipleValue %}
 
-{% macro individualDetailsFragment(form, nameLabel = 'Full name', addressHeading = 'Address', addressHint, hasCorrespondenceAddress = false) %}
-  {{ textInput(t(nameLabel), 'name', form, autocomplete = 'name') }}
+{% macro partyDetailsFragment(form, fields, addressSection, showCorrespondenceAddressSection = false) %}
+  {% for field in fields %}
+    {% if field.readOnly %}
+      {{ readOnlyField(field.label, field.value) }}
+    {% else %}
+      {{ textInput(form = form, label = t(field.label), name = field.name, autocomplete = field.autocomplete) }}
```

Comment:
I don't think t should be applied here, done inside the textInput instead

Reply:
A `input` does not translate labels itself. Something we should cleanup service wide, did not want to touch this now.